element_lab/lab.py

```diff
@@ -18,18 +18,42 @@ def activate(schema_name, create_schema=True, create_tables=True):
     )
 
 
+@schema
+class Organization(dj.Imported):
+    definition = """# Top-level list of all organizations involved in any of the projects
+    organization          : varchar(24)                # Abbreviated organization name
+    ---
+    organization_name          : varchar(255)   # full organization name
+    organization_address=''    : varchar(512)                # address of the organization
+    organization_comment=''    : varchar(1024)               # additional notes on the organization
+    """
+
+
 @schema
 class Lab(dj.Lookup):
     definition = """
-    lab             : varchar(24)    #  Abbreviated lab name
+    lab             : varchar(24)    # Abbreviated lab name
     ---
     lab_name        : varchar(255)   # full lab name
     institution     : varchar(255)
     address         : varchar(255)
-    time_zone       : varchar(64)    # 'UTC±X' format for NWB export
+    time_zone       : varchar(64)    # 'UTC±X' format for NWB export or timezone, e.g., America/New_York
```

Comment:
Do we plan to maintain timezone information on a per-lab or per-session basis? I proposed per-session [here](https://github.com/datajoint/element-session/pull/24/files). If we plan to go that route, I would remove here avoid redundancy.
If we plan on per-lab, accepting free input here could be really difficult to transform to UTC offset. 
```suggestion
    time_zone       : varchar(64)    # 'UTC±X' format or timezone, e.g., America/New_York
```

Reply:
`ZoneInfo` will throw an error if it can't find a timezone from a given key. Storing an offset will give wrong timezone information depending on when the datetime and zone is combined. 

```
import datetime as dt 
from zoneinfo import ZoneInfo

iana_db_name = "America/Chicago"
offset = dt.timezone(dt.timedelta(hours=-5))

dt_ = dt.datetime.fromisoformat("2022-08-29T00:00:00")
t_tz = dt_.replace(tzinfo=offset)
print(t_tz)
t_zi = dt_.replace(tzinfo=ZoneInfo(iana_db_name))
print(t_zi)
```

Output:

```
2022-08-29 00:00:00-05:00
2022-08-29 00:00:00-05:00
```

```
# daylight savings ends
dt_ = dt.datetime.fromisoformat("2022-11-07T00:00:00")
t_tz = dt_.replace(tzinfo=offset)
print(t_tz)
t_zi = dt_.replace(tzinfo=ZoneInfo(iana_db_name))
print(t_zi)
```

Output:

```
2022-11-07 00:00:00-05:00
2022-11-07 00:00:00-06:00
```

Reply:
I suggest that the time should always be stored in UTC. Timezone is a display preference and does not need to be stored.

Reply:
I can imagine a hypothetical in which user A stores data in NWB and uploads to DANDI. User B wants to reuse the data for a new analysis comparing morning and evening data collection. Without timezone information, User B might mistakenly assume the data was collected at UTC+0 rather than UTC-5

@iamamutt - Is there a clean way to enforce a valid value of timezone? ± local daylight savings is error, but perhaps a tolerable margin of error

Reply:
Either way, storing datetime as UTC doesn't solve the problem of not needing to know the timezone for the origin of the data to convert it to UTC. Also, I just realized ZoneInfo is 3.9 only. https://docs.python.org/3/library/zoneinfo.html#module-zoneinfo

Reply:
The data entry interface or the ingestion procedure should be responsible for convert the times into UTC.

Reply:
Yes, this is how I'm doing it, but using the timezone field as an optional method of getting the datetime back to the origin datetime in case time of day information may be needed, as in the situation @CBroz1 mentioned.